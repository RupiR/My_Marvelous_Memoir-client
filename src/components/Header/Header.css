.Header {
  height: 40px;
  display: flex;
  flex-direction: column;
}

.Header h1 {
  margin: auto;
  margin-bottom: 10px;
}

.Header h1 a {
  text-decoration: none;
  height: 100px;
  max-width: 600px;
  vertical-align: middle;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
}

.Header a {
  color: #062d30;
}

.Header__not-logged-in {
  display: flex;
  justify-content: center;
}

.Header__logged-in {
  display: flex;
  justify-content: center;
}

.Header__not-logged-in .Hyph {
  display: flex;
}

.computerlogo {
  height: 200px;
  width: 200px;
}

/* mobile view */

@media screen and (max-width: 450px) {
  header {
    align-items: baseline;
    padding: 0 !important;
  }
  .Header h1 a {
    padding: 0 !important;
  }
  .title {
    padding-left: 10px;
  }
}

/* desktop view */

@media screen and (min-width: 450px) {
  .Header h1 {
    font-size: 22pt;
  }

  .Header__not-logged-in {
    flex-direction: row;
  }

  .Header__not-logged-in .Hyph {
    display: initial;
  }

  .hamburgermenu {
    display: none;
  }
}
