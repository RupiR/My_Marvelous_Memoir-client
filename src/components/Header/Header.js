import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import IdleService from '../../services/idle-service'
import './Header.css'

export default class Header extends Component {
  state = {
    showMenu: window.screen.width < 500 ? false : true
  }
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    /* when logging out, clear the callbacks to the refresh api and idle auto logout */
    TokenService.clearCallbackBeforeExpiry()
    IdleService.unRegisterIdleResets()
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link
          to='/userpage'>
          Userpage
        </Link>
        &nbsp;&nbsp;&nbsp;
        <Link
          to='/posts'>
          Public Posts
        </Link>
        &nbsp;&nbsp;&nbsp;
        <Link
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='Header__not-logged-in'>
        <Link
          to='/register'>
          Register
        </Link>
        &nbsp;&nbsp;&nbsp;
        <Link
          to='/login'>
          Login
        </Link>
      </div>
    )
  }

  toggleMenu = () => {
    this.setState({
      showMenu: !this.state.showMenu
    })
  }

  render() {
    let menu
    if (this.state.showMenu) {
      menu = TokenService.hasAuthToken()
        ? this.renderLogoutLink()
        : this.renderLoginLink()
    }
    return (
      <nav className='Header'>
        <h1>
          <Link to='/'>
            <img className='computerlogo' alt="computerlogo" src='/assets/computerlogo.png' />
          </Link>
        </h1>

        {menu}
        <div className='hamburgermenu'>
          <button onClick={this.toggleMenu}>&#9776;</button>
        </div>
      </nav>
    )
  }
}
