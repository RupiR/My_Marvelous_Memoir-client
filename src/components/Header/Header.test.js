import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('<Header />', () => {
    it('Renders without crashing', () => {
        shallow(<Header />);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<Header />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});