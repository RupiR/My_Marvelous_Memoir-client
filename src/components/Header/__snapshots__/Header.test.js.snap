// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<nav
  className="Header"
>
  <h1>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="computerlogo"
        className="computerlogo"
        src="/assets/computerlogo.png"
      />
    </a>
  </h1>
  <div
    className="hamburgermenu"
  >
    <button
      onClick={[Function]}
    >
      â˜°
    </button>
  </div>
</nav>
`;
